### ISSUE TITLE
Running the program

### ISSUE THREAD

**Issue by JanardanS** [karansher/computer-graphics-raster-images#21]
I am a bit confused on how to run the program exactly. Aftering doing cmake .. to the build directory, and running ./raster.sln, what occurs after? Do we have to build the solution? After do we start with debugging. If so, it doesnt seem to work as it gives me an error message saying system cannot find file specified.

**Comment by abhimadan** [comment]
You don't run the sln file, since it's a Visual Studio project file. You open the .sln file in Visual Studio, then compile and run your program.

**Comment by JanardanS** [comment]
So I think i have the right idea then, because I did build for the raster project, then i did \Debug\raster, but where do I see the ppm files. Do i just open through file explorer, how can I view it exactly?

**Comment by abhimadan** [comment]
You should see them in your `build` folder, and then you should be able to open them with a image viewer app. I'm not sure if Windows's default image viewer app works, but if it doesn't, you can try qview: https://interversehq.com/qview/

**Comment by RamDaj** [comment]
Do we need to do `cmake ..` every time we modify the code? Do we have to rebuild the raster.sln file every time? It seems really inconvenient as I also have to make the raster project the default instead of ALL_BUILD every time

**Comment by abhimadan** [comment]
No, you just need to run it once. After you have a working raster.sln, you just use Visual Studio.

**Comment by RamDaj** [comment]
By using visual studio, do you mean clicking the "Start without debugging" button on the `main.cpp` file, or by running `.\Debug\raster` (I'm on windows)? when I run using the debugger, the value of rgba is null.

**Comment by abhimadan** [comment]
To avoid confusion, let's break down the steps that you need to take from the time you first clone the assignment repo to when you run the program.

1. Generating the platform-specific build files. This is accomplished through running `cmake ..`. Once you've successfully completed this step, you don't need to run it again.
2. Compiling your code. On Windows, it's easiest to do this through the Visual Studio IDE. You can compile your code without running it by pressing ctrl+shift+b.
3. Running your code. If you use the "start without debugging" option, this is combined with step 2, but if you want to run your program on the command line, you use `.\Debug\raster`.

So even if you want to run your program via the command line, you still need Visual Studio for compilation.

I'd recommend figuring out your IDE issues since the debugger is a very useful tool. In this particular case, you need to make sure that the `raster` project's working directory is the solution directory. See [this post](https://stackoverflow.com/questions/4815423/how-do-i-set-the-working-directory-to-the-solution-directory) for instructions on how to do so (though you just want it to be `$(SolutionDir)` and not configuration-specific).
